include/seqan3/search/detail/policy_search_result_builder.hpp

```diff
@@ -114,40 +75,76 @@ struct policy_search_result_builder
-    //!\cond
-        requires search_traits_type::search_return_text_position &&
+        requires search_traits_type::output_requires_locate_call &&
                  (!search_traits_type::search_single_best_hit)
-    //!\endcond
     void make_results(std::vector<index_cursor_t> internal_hits, query_index_t idx, callback_t && callback)
     {
-        using ref_location_pair_t = decltype(std::declval<index_cursor_t>().lazy_locate().front());
-        static_assert(std::destructible<decltype(search_result_type{std::declval<query_index_t &&>(),
-                                                                    std::declval<ref_location_pair_t &>().first,
-                                                                    std::declval<ref_location_pair_t &>().second})>,
-                      "The configured search result type is not compatible with the requested search output. "
-                      "Note trying to instantiate search result with text positions with either all, all_best or "
-                      "strata hit strategy.");
-
         std::vector<search_result_type> results{};
         results.reserve(internal_hits.size()); // expect at least as many text positions as cursors, possibly more
 
-        for (auto const & cursor : internal_hits)
-            for (auto && [ref_id, ref_pos] : cursor.locate())
-                results.push_back(search_result_type{std::move(idx), ref_id, ref_pos});
+        make_results_impl(std::move(internal_hits), idx, [&results](auto && search_result)
+        {
+            results.push_back(std::move(search_result));
+        });
 
         // sort by reference id or by reference position if both have the same reference id.
-        auto compare = [] (auto const & r1, auto const & r2)
+        std::sort(results.begin(), results.end(), [] (auto const & r1, auto const & r2)
         {
             return (r1.reference_id() == r2.reference_id()) ? (r1.reference_begin_pos() < r2.reference_begin_pos())
                                                             : (r1.reference_id() < r2.reference_id());
-        };
+        });
 
-        std::sort(results.begin(), results.end(), compare);
         results.erase(std::unique(results.begin(), results.end()), results.end());
 
         for (auto && search_result : results)
             callback(std::move(search_result));
     }
+
+private:
+    /*!\brief Invokes the callback on each seqan3::search_result and calls locate on the cursor depending on the config.
+     *
+     * \tparam index_cursor_t The type of index cursor used in the search algorithm.
+     * \tparam query_index_t The index type of the query.
+     * \tparam callback_t The callback which is called for every hit.
+     *
+     * \param[in] internal_hits internal_hits A range over internal cursor results.
+     * \param[in] idx The index associated with the current query.
+     * \param[in] callback The callback to invoke for every hit.
+     */
```

Comment:
It would be great to have a details section that explains a little what the function does.

Reply:
```suggestion
     *
     * \details
     *
     * For each cursor `in internal_hits`, this function calls `cursor.layz_locate()` if the search configuration
     * requires it (search_traits_type::output_requires_locate_call) and then constructs a seqan3::search_result from
     * the resulting data. The seqan3::search_result will be filled only with the data that was asked for by the user
     * via the `search_traits_type::output_[...]` trait (e.g. `search_traits_type::output_query_id`).
     */
```